oak_attestation_verification/src/verifier.rs

```diff
@@ -760,22 +761,39 @@ fn verify_hex_digests(actual: &HexDigest, expected: &HexDigest) -> anyhow::Resul
     }
 }
 
-fn verify_regex(actual: &str, expected: &RegexReferenceValue) -> anyhow::Result<()> {
+fn verify_text(actual: &str, expected: &TextReferenceValue) -> anyhow::Result<()> {
     match expected.r#type.as_ref() {
-        Some(regex_reference_value::Type::Skip(_)) => Ok(()),
-        Some(regex_reference_value::Type::Regex(regex)) => {
+        Some(text_reference_value::Type::Skip(_)) => Ok(()),
+        Some(text_reference_value::Type::Regex(regex)) => verify_regex(actual, regex),
+        Some(text_reference_value::Type::StringLiteral(string_literal)) => Ok(anyhow::ensure!(
+            actual == string_literal.value,
+            format!("value doesn't match the reference value string literal: {actual}")
+        )),
+        None => Err(anyhow::anyhow!("missing skip or value in the text reference value")),
+    }
+}
+
+fn verify_regex(
+    actual: &str,
+    regex: &oak_proto_rust::oak::attestation::v1::Regex,
+) -> anyhow::Result<()> {
+    if cfg!(feature = "regex") {
+        #[cfg(feature = "regex")]
```

Comment:
are both config guards needed?

Reply:
#[cfg(feature = "regex")] allows using the Regex type only when the feature is enabled, the if cfg!(feature = "regex") condition prevents the unreachable code warning for returning the error - but perhaps there is a more elegant way to do it?

Reply:
I think another option could be to define two verify_regex functions, one for when the feature is on, one for when it is off. Something like:

```
#[cfg(feature = "regex")]
fn verify_regex(
    actual: &str,
    regex: &oak_proto_rust::oak::attestation::v1::Regex,
) -> anyhow::Result<()> {
  // the real implementation
}

#[cfg(not(feature = "regex"))]
fn verify_regex(
    _actual: &str,
    _regex: &oak_proto_rust::oak::attestation::v1::Regex,
) -> anyhow::Result<()> {
  Err(anyhow::anyhow!("verification of regex values not supported"))
}
```

Reply:
Oh, that's really nice, thanks!